### PR TITLE
Inlined version of recognized methods

### PR DESCRIPTION
**Pull request by sahithiponangi**
java_nio_Bits_keepAlive and
 java_lang_ref_Reference_reachabilityFence have been added
 by consulting the semantics on other architectures.

Closes: #7137

Signed-off-by: Siri Sahithi Ponangi <sahithi.ponangi@unb.ca>

### CODE REVIEW

File: runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp [hunk 1]
```diff
@@ -1693,6 +1693,19 @@ J9::ARM64::CodeGenerator::inlineDirectCall(TR::Node *node, TR::Register *&result
       {
       switch (methodSymbol->getRecognizedMethod())
          {
+         case TR::java_nio_Bits_keepAlive:
+         case TR::java_lang_ref_Reference_reachabilityFence:
+            {
+            TR::Node *paramNode = node->getFirstChild();
```
Comment (0xdaryl) [inline runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp:1699]: If you copied this implementation from x86, can you copy the comments that were in that code as well?  It helps to explain what this code is trying to achieve.

File: runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp [hunk 2]
```diff
@@ -1928,6 +1928,28 @@ J9::ARM64::CodeGenerator::inlineDirectCall(TR::Node *node, TR::Register *&result
       {
       switch (methodSymbol->getRecognizedMethod())
          {
+         case TR::java_nio_Bits_keepAlive:
+         // The only purpose of keepAlive is to prevent an otherwise
+         // unreachable object from being garbage collected, because we don't
+         // want its finalizer to be called too early.  There's no need to
+         // generate a full-blown call site just for this purpose.
+         case TR::java_lang_ref_Reference_reachabilityFence:
```
Comment (knn-k) [inline runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp:1936]: There is a trailing space at the end of this line.

**Comment by knn-k** [comment]
jenkins test sanity alinux64 jdk11

**Comment by 0xdaryl** [comment]
jenkins test sanity alinux64 jdk11

Previous checks passed, but re-running in 2021.

**Comment by knn-k** [comment]
jenkins test sanity alinux64 jdk11

File: runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp (hunk 1 above)
Comment (0xdaryl) [reply runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp:1988]: I find the placement of these comments to be a bit odd.  Can you move them down below the opening brace like x86 has them?  Also, remove the specific reference to `keepAlive` as this applies to a couple of different recognized methods.  Maybe say "these functions" instead?

**Review by 0xdaryl — CHANGES_REQUESTED** [review]
Minor comment about comment content and placement.  Code otherwise is fine.

**Comment by 0xdaryl** [comment]
jenkins test sanity alinux64 jdk11

**Review by 0xdaryl — APPROVED** [review]

**Pull request merged**